catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -195,7 +212,36 @@ public void maskId(String id) {
 
     @Override
     public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
-        return client.query(request);
+        SourceResponse response = client.query(request);
+        queryPendingNrtIndex(request, response);
+        return response;
+    }
+
+    private void queryPendingNrtIndex(QueryRequest request, SourceResponse response)
+            throws UnsupportedQueryException {
+        if (request == null || request.getQuery() == null) {
+            return;
+        }
+
+        Set<String> ids = filterAdapter.adapt(request.getQuery(),
+                new MetacardIdEqualityFilterDelegate());
+        if (ids.size() == 0 || ids.size() <= response.getResults()
```

Comment:
Are you sure that ids.size() being less than the number of results will always be a valid short circuit? 
i'm thinking something like
`id = t1 AND metacard.version.bla = somethingelse` Where the second condition matches 5 some odd metacards, so the results size is 5, but the ids size is 1.. since its less it short circuits. Though wouldn't we want to still fill in the one metacard id that was requested?

Or am I simply mislead and does this somehow not get executed if there is anything other than an ID query?